src/frontend/org/voltdb/calciteadapter/rules/VoltDBRules.java

```diff
@@ -83,20 +86,36 @@
             , VoltDBSortConvertRule.INSTANCE_NONE
             , VoltDBSortConvertRule.INSTANCE_VOLTDB
             , VoltDBPhysicalLimitRule.INSTANCE
+            , VoltDBPhysicalAggregateRule.INSTANCE
+    };
 
-            );
-
-    public static Program RULES_SET_2 = Programs.ofRules(
+    public static RelOptRule[] HEP_RULES = {
             // VoltDB Inline Rules
 
             VoltDBCalcScanMergeRule.INSTANCE
             , VoltDBLimitScanMergeRule.INSTANCE
             , VoltDBLimitSortMergeRule.INSTANCE
+            , VoltDBCalcAggregateMergeRule.INSTANCE
+    };
+
+    public static Program VOLCANO_PROGRAM_0 = Programs.ofRules(
+            VOLCANO_RULES_0
+            );
 
+    public static Program VOLCANO_PROGRAM_1 = Programs.ofRules(
+            VOLCANO_RULES_1
             );
 
-    public static Program[] getProgram() {
+    public static Program HEP_PROGRAM = Programs.ofRules(
+            HEP_RULES
+            );
 
-        return new Program[] {RULES_SET_0, RULES_SET_1, RULES_SET_2};
+    public static Program[] getVolcanoPrograms() {
+        return new Program[] {VOLCANO_PROGRAM_0, VOLCANO_PROGRAM_1};
     }
+
+    public static Program getHepProgram() {
```

Comment:
I think maybe instead of describing the kind of planner in the code, it would be better to describe the phase of planning being performed.  So this would be the `INLINING_PROGRAM`, and the corresponding rules could be in the `rules.inlining` package and so on.

Reply:
Agree. Renamed.